src/node_sockaddr.h

```diff
@@ -126,10 +126,10 @@ class SocketAddress : public MemoryRetainer {
   inline void Update(uint8_t* data, size_t len);
   inline void Update(const sockaddr* data, size_t len);
 
-  static SocketAddress FromSockName(const uv_udp_t& handle);
-  static SocketAddress FromSockName(const uv_tcp_t& handle);
-  static SocketAddress FromPeerName(const uv_udp_t& handle);
-  static SocketAddress FromPeerName(const uv_tcp_t& handle);
+  static std::shared_ptr<SocketAddress> FromSockName(const uv_udp_t& handle);
+  static std::shared_ptr<SocketAddress> FromSockName(const uv_tcp_t& handle);
+  static std::shared_ptr<SocketAddress> FromPeerName(const uv_udp_t& handle);
+  static std::shared_ptr<SocketAddress> FromPeerName(const uv_tcp_t& handle);
```

Comment:
Can you explain why? `SocketAddress` is copyable, and copying it would be expected to have lower performance overhead than a separate allocation

(And even if not ... why `std::shared_ptr` instead of `std::unique_ptr`?)